app/views/criteria/_new_criterion.html.erb

```diff
@@ -1,8 +1,8 @@
 <li id='new_criterion'>
 	<%= form_for @criterion,
-    url: { controller: "#{criterion_type.to_s.tableize}",
-           action: 'index',
-           assignment_id: assignment.id },
+    url: assignment_criteria_path(assignment_id: assignment.id,
+                                  criterion_type: @criterion_type),
```

Comment:
I was debating whether to pass the argument or get the name with .class.to_s whenever I can. I will avoid passing this argument unless it is strictly necessary. Same for below.